jenkins/PerfCI_Grafana_Deployment/Jenkinsfile

```diff
@@ -69,74 +69,91 @@ pipeline {
             }
         }
 
-        stage('Check for changes in dashboard.json and main.libsonnet') {
-        steps {
-            script {
-                CHANGES_DASHBOARD = sh(script: "git diff --quiet ${WORKSPACE}/benchmark_runner/grafana/perf/dashboard.json || echo true", returnStatus: true) == 0
-                echo "CHANGES_DASHBOARD=${CHANGES_DASHBOARD}"
-
-                CHANGES_MAIN_LIBSONNET = sh(script: "git diff --quiet ${WORKSPACE}/benchmark_runner/grafana/perf/jsonnet/main.libsonnet || echo true", returnStatus: true) == 0
-                echo "CHANGES_MAIN_LIBSONNET=${CHANGES_MAIN_LIBSONNET}"
+        stage('Check for changes in main.libsonnet') {
+            steps {
+                script {
+                    def file_path = "${WORKSPACE}/benchmark_runner/grafana/perf/jsonnet/main.libsonnet"
+                    // Check if the file exists
+                    if (fileExists(file_path)) {
+                        // Check if there are any changes in the file
+                        CHANGES_MAIN_LIBSONNET = sh(script: "git diff --quiet $file_path", returnStatus: true) != 0
+                        echo "CHANGES_MAIN_LIBSONNET=${CHANGES_MAIN_LIBSONNET}"
+                    } else {
+                        echo "File $file_path does not exist"
```

Comment:
Can (and should) you signal an error in this case?

Reply:
What do you mean ?

Reply:
Fixed